systori/static/ts/src/project_list.ts

```diff
@@ -1,144 +1,55 @@
-import { ArrayListMultimap, Multimap } from "@systori/lib/multimap";
-import naturalSort from "natural-sort";
-
-// const searchRequestFired = false;
-// const attemptMade = false;
-// let searchMatches: Array<number>;
-let phaseFilter: Array<string>;
+const filterBar = document.querySelector("#filter-bar");
+const tileContainer = document.querySelector("#tile-container");
 enum PhaseOrder {
-    prospective = "prospective",
-    tendering = "tendering",
-    planning = "planning",
-    executing = "executing",
-    settlement = "settlement",
-    warranty = "warranty",
-    finished = "finished",
+    prospective,
+    tendering,
+    planning,
+    executing,
+    settlement,
+    warranty,
+    finished,
 }
 
-function sortProjects(e: Event): void {
-    if (e == null) return;
-    let lookup: Map<string | number, SystoriProjectTile> = new Map();
-    // const i = 0;
-
-    const btn = e.target as SystoriSortButton;
-    btn.activateExclusive();
-
-    if (btn.type == "id") {
-        Array.from(document.querySelectorAll<SystoriProjectTile>(".tile")).map(
-            e => lookup.set(parseInt(e.dataset["pk"] || "0"), e),
-        );
-    } else if (btn.type == "name") {
-        Array.from(document.querySelectorAll<SystoriProjectTile>(".tile")).map(
-            e => lookup.set(e.dataset["name"] || "", e),
-        );
-    } else if (btn.type == "phase") {
-        const lookup2: Multimap<string, HTMLElement> = new ArrayListMultimap();
-        lookup = new Map();
-        Array.from(document.querySelectorAll<SystoriProjectTile>(".tile")).map(
-            e => lookup2.put(e.dataset["phase"] || "", e),
-        );
-        for (const key in PhaseOrder) {
-            // for (let element of lookup2[key]) {
-            //     console.log(element);
-            // }
-            console.log(key);
-        }
+class SystoriProjectTile extends HTMLDivElement {
+    constructor() {
+        super();
     }
-
-    let sortedKeys: Array<string | number> = Array.from(lookup.keys()).sort(
-        naturalSort(),
-    );
-    if (btn.reversed == true) {
-        sortedKeys = sortedKeys.reverse();
-        btn.reversed = false;
-    } else if (btn.reversed == false) {
-        btn.reversed = true;
+    get pk(): number {
+        return Number(this.dataset["pk"] || "0");
```

Comment:
```suggestion
        return parseInt(this.dataset["pk"]!);
```

Because `Nubmer()` does type conversion and `parseInt` actually parses it.